packages/@aws-cdk/aws-appsync/README.md

```diff
@@ -157,7 +157,7 @@ rdsDS.createResolver({
   }
   `),
   responseMappingTemplate: MappingTemplate.fromString(`
-    $util.rds.toJsonObject($ctx.result)
+    $utils.toJson($utils.rds.toJsonObject($ctx.result)[1][0])
```

Comment:
Can you add some explanation for these indices? Also, is there another solution that doesn't involve indices at all? (Using `Array.filter()`, for example).

Reply:
I think it depends on the sql statement you are running. In this case(taking directly from the doc), we are trying to run this `SELECT * WHERE id = :id`, which will return a list from db, hence we need to do the indices`[1][0]`. However, with a different sql, `SELECT DISTINCT * WHERE id = :id`, we can cut the indices to `$utils.toJson($utils.rds.toJsonObject($ctx.result)[0])` but we still need at least one indice. if you want to remove the need for indices at all, I think this would be a job for the vtl helper class. Most importantly, the vtl in the doc did not work and needs fix

Reply:
@otaviomacedo one would expect `$utils.rds.toJsonObject($ctx.result)` to correctly return the correct object, so that user wouldn't need to do the indices`[1]`. I don't have a proper way to test vtl, but what is in the index[0]? if in the vtl helper class we can remove whatever is in index 0, we can probably remove the usage of indices

Reply:
also, @otaviomacedo Have you tested whatever is in the doc? The example in the doc uses mysql, but mine uses postgresql. There shouldn't be any difference but I want to make sure

Reply:
Ok, I hadn't seen that in the [AppSync Tutorial] they also use these indices without much in the way of an explanation. So we can leave it like this here, too.

Just FYI, in the mutation the first index refers to the statement you provided in the request mapping template. Since we want the result of the query, which is the second element of the `statements` array, we pass the index 1. And the result itself is a list of elements. But in this case we know this list has only one element (because we are querying by ID), so we select that element by passing the index 0.

[AppSync Tutorial]: https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-rds-resolvers.html